contracts/RIFScheduler.sol

```diff
@@ -339,11 +343,13 @@ contract RIFScheduler is IERC677TransferReceiver, ReentrancyGuard, Pausable {
     }
   }
 
-  function multicall(bytes[] calldata data) external whenNotPaused returns (bytes[] memory results) {
+  // This function enables multiple contact call, it revertIfFails is false the execution will continue even if a transaction reverted.
+  // Otherwise this transaction will revert if any transaction reverts
+  function multicall(bytes[] calldata data, bool revertIfFails) external whenNotPaused returns (bytes[] memory results) {
     results = new bytes[](data.length);
     for (uint256 i = 0; i < data.length; i++) {
       (bool success, bytes memory result) = address(this).delegatecall(data[i]);
-      require(success);
+      require(success || !revertIfFails, string(abi.encodePacked('Transaction failed:', i.toString())));
       results[i] = result;
     }
     return results;
```

Comment:
Three things to note here:

- if `revertIfFails = false`, there might be a possibility that the payee doesn't get his money for the execution, but thats up for him to decide that, if he's ok with it... But we need to document that properly and a dangerous thing for him
- Because of the point above, only payee should be able to call `multicall`
- By default the SDK and/or service should use `revertIfFails = false` and fallback to execute each transaction individually

Reply:
@luisfontes19 are you suggesting a change, or documenting all this cases is ok?

Reply:
just documenting it